daemon/cluster/executor/container/container.go

```diff
@@ -186,13 +187,16 @@ func (c *containerConfig) exposedPorts() map[nat.Port]struct{} {
 }
 
 func (c *containerConfig) config() *enginecontainer.Config {
+	genericEnvs := genericresource.EnvFormat(c.task.AssignedGenericResources, "DOCKER_RESOURCE")
+	env := append(c.spec().Env, genericEnvs...)
+
 	config := &enginecontainer.Config{
 		Labels:       c.labels(),
 		StopSignal:   c.spec().StopSignal,
 		Tty:          c.spec().TTY,
 		OpenStdin:    c.spec().OpenStdin,
 		User:         c.spec().User,
-		Env:          c.spec().Env,
+		Env:          env,
```

Comment:
Isolation, or resource allocation (for example gpu devices) is a totally different scope :)
This PR is about resource accounting at the scheduler level and giving users the knobs.

The problem of GPUs in containers is not an easy one and right now we are solving it with a library that can be called at the runc level: https://github.com/NVIDIA/libnvidia-container